### ISSUE TITLE
checks required for words do not occur in the feedback documents  in query expansion-based retrieval 

### ISSUE THREAD

**Issue by 9meo** [terrier-org/terrier-core#145]
Problem: JavaException: JVM exception occurred: java.lang.NullPointerException when using the following model (Bo1 reweighing only)

reweighted_only = pt.BatchRetrieve(index, wmodel="DPH") >> pt.rewrite.Bo1QueryExpansion(index, fb_terms=0)

Problem Trace:
org.terrier.querying.DFRBagExpansionTerms.getExpandedTerms(DFRBagExpansionTerms.java:186)
org.terrier.querying.DFRBagExpansionTerms.getExpandedTerms(DFRBagExpansionTerms.java:156)
org.terrier.querying.QueryExpansion.expandQuery(QueryExpansion.java:207)

Possible reason: 
- the "query" contains words that do not occur in the feedback documents
- already check for the size of retrieved documents with the DPH model > 0

**Comment by cmacdonald** [comment]
Test from PyTerrier:
```python
import pyterrier as pt
if not pt.started():
  pt.init(logging="DEBUG", version="5.4-SNAPSHOT")

import pandas as pd
docs_df = pd.DataFrame([
    ["d1", "this is the first document of many documents"],
    ["d2", "this is another document"],
    ["d3", "the topic of this document is unknown"],
    ["d4", "the topic of this document is techniques and liquid"]

], columns=["docno", "text"])

indexer = pt.DFIndexer("./sample_index/", overwrite=True)
indexref = indexer.index(docs_df["text"], docs_df["docno"])
indexref.toString()

index = pt.IndexFactory.of(indexref)

print(index.getCollectionStatistics().toString())

topics = pt.get_dataset("vaswani").get_topics().head(1)
topics = pd.concat([topics]*3)
topics["docno"] = ["d1", "d2", "d3"]
topics["docid"] = [0, 1, 2]


from jnius import JavaException
reweight = pt.BatchRetrieve(index, wmodel="DPH") >> pt.rewrite.Bo1QueryExpansion(index, fb_terms=0)
try:
  print(reweight(topics).iloc[0].query)
except JavaException as ja:
  print("\n\t".join(ja.stacktrace))
  raise ja
```

**Comment by cmacdonald** [comment]
Now addressed for 5.4 release. Thanks for your help with this Sarawoot.
